From: John Spray <john.spray@redhat.com>
Date: Fri, 30 Sep 2016 21:57:35 +0100
Subject: mds: make mds_role_t member order match constructor

Previously this was mildy confusing.

Signed-off-by: John Spray <john.spray@redhat.com>
(cherry picked from commit 63d711d4e9decd2940c55aa697e6a588593da23a)

Resolves: rhbz#1384002
---
 src/mds/mdstypes.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/mds/mdstypes.h b/src/mds/mdstypes.h
index b4195eb..98a732e 100644
--- a/src/mds/mdstypes.h
+++ b/src/mds/mdstypes.h
@@ -90,13 +90,14 @@ extern const mds_rank_t MDS_RANK_NONE;
 class mds_role_t
 {
   public:
-  mds_rank_t rank;
   fs_cluster_id_t fscid;
+  mds_rank_t rank;
+
   mds_role_t(fs_cluster_id_t fscid_, mds_rank_t rank_)
-    : rank(rank_), fscid(fscid_)
+    : fscid(fscid_), rank(rank_)
   {}
   mds_role_t()
-    : rank(MDS_RANK_NONE), fscid(FS_CLUSTER_ID_NONE)
+    : fscid(FS_CLUSTER_ID_NONE), rank(MDS_RANK_NONE)
   {}
   bool operator<(mds_role_t const &rhs) const
   {
